packages/manager/src/layouts/OAuth.tsx

```diff
@@ -166,18 +136,29 @@ export class OAuthCallbackPage extends Component<CombinedProps, State> {
     } catch (error) {
       clearStorageAndRedirectToLogout();
     }
-  }
+  };
+
+  React.useEffect(() => {
+    if (!location.search || location.search.length < 2) {
+      clearStorageAndRedirectToLogout();
+      return;
+    }
 
-  render() {
-    const { isLoading } = this.state;
+    const { code, returnTo, state: nonce } = getQueryParamsFromQueryString(
+      location.search
+    );
 
-    if (isLoading) {
-      return <SplashScreen />;
+    if (!code || !returnTo || !nonce) {
+      clearStorageAndRedirectToLogout();
+      return;
     }
 
-    return null;
-  }
-}
+    exchangeAuthorizationCodeForToken(code, returnTo, nonce);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [location.search, dispatchStartSession, history]);
+
+  return isLoading ? <SplashScreen /> : null;
```

Comment:
If we are on this page, cloud manager is authenticating, so we can probably just return the `SplashScreen` at all times. I don't think a loading state is even necessary. 

```suggestion
  return <SplashScreen />
```